rest/client/rest.go

```diff
@@ -260,3 +261,24 @@ func (c *communicatorImpl) GetCurrentUsersKeys(ctx context.Context) ([]model.API
 
 	return keys, nil
 }
+
+func (c *communicatorImpl) AddPublicKey(ctx context.Context, keyName, keyValue string) error {
+	info := requestInfo{
+		method:  post,
+		version: apiVersion2,
+		path:    "keys",
+	}
+
+	key := model.APIPubKey{
+		Name: model.APIString(keyName),
+		Key:  model.APIString(keyValue),
+	}
+
+	resp, client_err := c.request(ctx, info, key)
+	defer resp.Body.Close()
+	if client_err != nil || resp.StatusCode != http.StatusOK {
```

Comment:
just name the variable "err"

if err != nil, then you should close the body, and handle non-200 responses separately